src/Http/Requests/DestroyThread.php

```diff
@@ -24,17 +26,38 @@ public function fulfill()
     {
         $thread = $this->route('thread');
 
-        if (config('forum.general.soft_deletes') && isset($this->validated()['permadelete']) && $this->validated()['permadelete'] && method_exists($thread, 'forceDelete'))
+        $thread->readers()->detach();
+        
+        $threadIsTrashed = $thread->trashed();
+
+        if ($this->isPermaDeleting() && method_exists($thread, 'forceDelete'))
         {
+            $thread->posts()->withTrashed()->forceDelete();
             $thread->forceDelete();
         }
         else
         {
             $thread->delete();
         }
 
-        $thread->category->syncCurrentThreads();
+        $category = $thread->category;
+
+        if (! $threadIsTrashed)
+        {
+            // Only change category stats and FKs if the thread wasn't already soft-deleted
+            $category->update([
+                'newest_thread_id' => $category->getNewestThreadId(),
```

Comment:
Could potentially just updated `newest_thread_id` and `latest_active_thread_id` only if they match the `thread_id`? This could save 2 unnecessary queries in some cases.